package tests;

import io.restassured.RestAssured;
import io.restassured.response.Response;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;
import utils.XLUtility;

public class GetAllUsersTest extends BaseTest {

    @DataProvider(name = "UserData")
    public Object[][] getData() throws Exception {
        return XLUtility.getTestData("Tests");
    }

    @Test(dataProvider = "UserData")
    public void testGetAllUsers(String testType, String testName, String baseUrl,
                                String endpoint, String username, String password,
                                String expectedResult, String status, int rowIndex) throws Exception {

        if (!"GetAllUsers".equalsIgnoreCase(testType)) return; // skip other tests

        System.out.println("Running GetAllUsers test -> " + testName);

        String token = getAuthToken(baseUrl, username, password);

        Response response = RestAssured.given()
                .header("Authorization", "Bearer " + token)
                .get(baseUrl + endpoint);

        System.out.println("Response: " + response.asString());

        try {
            if ("PASS".equalsIgnoreCase(expectedResult)) {
                response.then().statusCode(200);
                XLUtility.setCellData("Tests", rowIndex, 7, "PASS");
            } else {
                response.then().statusCode(403);
                XLUtility.setCellData("Tests", rowIndex, 7, "PASS");
            }
        } catch (AssertionError e) {
            XLUtility.setCellData("Tests", rowIndex, 7, "FAIL");
            throw e;
        }
    }
}
